code/game/machinery/computer/atmos_control.dm

```diff
@@ -418,54 +389,51 @@ var/global/list/atmos_controllers = list()
 	if(href_list["set_preset_setting"])
 		if(!(log_in_id && (access_ce in log_in_id.GetAccess()) || emagged || usr.hasFullAccess()))
 			return 1
-		switch(href_list["set_preset_setting"])
-			if("oxygen", "nitrogen", "carbon_dioxide", "plasma", "n2o", "other", "pressure", "temperature")
-				//this could be better...
-				var/selected = null
+		if(href_list["set_preset_setting"] == "target_temperature")
+			var/max_temperature = MAX_TARGET_TEMPERATURE - T0C //these defines should come from code\game\machinery\alarm.dm
+			var/min_temperature = MIN_TARGET_TEMPERATURE - T0C
+			var/input_temperature = input("What temperature (in C) would you like the system to target? (Capped between [min_temperature]C and [max_temperature]C).\n\nNote that the cooling unit in this air alarm can not go below [MIN_TEMPERATURE]C or above [MAX_TEMPERATURE]C by itself. ", "Thermostat Controls") as num|null
+			if(input_temperature==null)
+				return 1
+			if(!input_temperature || input_temperature >= max_temperature || input_temperature <= min_temperature)
+				to_chat(usr, "<span class='warning'>Temperature must be between [min_temperature]C and [max_temperature]C.</span>")
+			else
+				input_temperature = input_temperature + T0C
+			selected_preset.target_temperature = input_temperature
+			return 1
+		else if(href_list["set_preset_setting"] == "scrubbed_gases")
+			var/gas = href_list["gas"]
+			if(gas)
+				// This "toggle" isn't greatest for performance... but it should only occur once per user input.
+				if(selected_preset.scrubbed_gases.Find(gas))
+					selected_preset.scrubbed_gases -= gas
+				else
+					selected_preset.scrubbed_gases += gas
```

Comment:
You can change it to the same thing I mentioned a few comments ago for slightly better performance